src/TC_util.cpp

```diff
@@ -0,0 +1,58 @@
+#include "TC_util.h"
+
+#include <string.h>
+
+#include "Devices/Serial_TC.h"
+
+// Example: strscpy(buffer, source, sizeof(buffer));
+int strscpy(char *destination, const char *source, unsigned long sizeOfDestination) {
+  unsigned long sourceLength = strnlen(source, sizeOfDestination);
+  if (sourceLength < sizeOfDestination) {
+    // Put a null terminator in the final byte
+    *((char *)memcpy(destination, source, sourceLength) + sourceLength) = '\0';
+    return 0;
+  } else {
+    // Put a null terminator in the final byte
+    *((char *)memcpy(destination, source, sizeOfDestination - 1) + sizeOfDestination - 1) = '\0';
+    serial(F("WARNING! String \"%s\" was truncated to \"%s\""), source, destination);
+    // TODO: Log a WARNING that a string was truncated
```

Comment:
I'd prefer there to be a separate notification process for warnings rather than hoping that someday someone reads the log files carefully. Maybe send e-mails or text messages to interested parties if that were possible.